File: programmers/기본연습문제/정수 제곱근 판별.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    
    long long a = sqrt(n);
    if (a*a == n) {	// 양의 정수x의 제곱인 경우
        answer=pow(a+1,2);
    }
    else {	// 아닌 경우
        answer=-1;
    }
    
    return answer;
}

File: python/수학/정수 제곱근 판별.py
#https://school.programmers.co.kr/learn/courses/30/lessons/12934?language=python3

from math import sqrt

def solution(n):
    
    if sqrt(n) % 1:     #소수점 아래 자리만 남게 됨(소수점이 있음 = 완전제곱수가 아님)
        answer = -1
    else:
        answer = (sqrt(n)+1)**2
        
    return answer